flows/actions/send_msg.go

```diff
@@ -150,27 +151,32 @@ func (a *SendMsgAction) getTemplateMsg(run flows.Run, urn urns.URN, channelRef *
 	}
 
 	// the message we return is an approximate preview of what the channel will send using the template
-	var previewParts []string
+	var previewText []string
+	var previewAttachments []utils.Attachment
 	var previewQRs []string
 
 	for _, comp := range translation.Components() {
 		previewContent := comp.Content()
 		for key, index := range comp.Variables() {
-			previewContent = strings.ReplaceAll(previewContent, fmt.Sprintf("{{%s}}", key), variables[index].Value)
+			variable := variables[index]
+			if variable.Type == "text" {
+				previewContent = strings.ReplaceAll(previewContent, fmt.Sprintf("{{%s}}", key), variable.Value)
+			} else if variable.Type == "image" && variable.Value != "" {
```

Comment:
What about videos or documents? which are the other type that can be supported

Reply:
Yeah good point we should make this work for all header types since we're doing this work